src/lib/ivy/compile-source-files.ts

```diff
@@ -0,0 +1,168 @@
+import * as ng from '@angular/compiler-cli';
+import { OptimizeFor } from '@angular/compiler-cli/src/ngtsc/typecheck/api';
+import { NgtscProgram } from '@angular/compiler-cli/src/ngtsc/program';
+import * as ts from 'typescript';
+import * as log from '../utils/log';
+import * as ivy from '../ivy';
+import { augmentProgramWithVersioning, cacheCompilerHost } from './cache-compiler-host';
+import { BuildGraph } from '../graph/build-graph';
+import { EntryPointNode, isEntryPointInProgress } from '../ng-package/nodes';
+import { NgccProcessor } from '../ngc/ngcc-processor';
+import { ngccTransformCompilerHost } from '../ts/ngcc-transform-compiler-host';
+
+export async function compileSourceFiles(
+  graph: BuildGraph,
+  tsConfig: ng.ParsedConfiguration,
+  moduleResolutionCache: ts.ModuleResolutionCache,
+  extraOptions?: Partial<ng.CompilerOptions>,
+  stylesheetProcessor?: ivy.StylesheetProcessor,
+  ngccProcessor?: NgccProcessor,
+  watch?: boolean,
+) {
+  log.debug(`ngc (v${ng.VERSION.full})`);
+
+  const tsConfigOptions: ng.CompilerOptions = { ...tsConfig.options, ...extraOptions };
+  const entryPoint: EntryPointNode = graph.find(isEntryPointInProgress());
+
+  let tsCompilerHost = cacheCompilerHost(
+    graph,
+    entryPoint,
+    tsConfigOptions,
+    moduleResolutionCache,
+    stylesheetProcessor,
+  );
+
+  if (tsConfigOptions.enableIvy && ngccProcessor) {
```

Comment:
This tests for `enableIvy`, but then below an `NgtscProgram` is always created which will therefore always be Ivy.